src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAssessmentDao.java

```diff
@@ -0,0 +1,154 @@
+
+import static java.util.stream.Collectors.toList;
+import static org.sagebionetworks.bridge.BridgeConstants.SHARED_STUDY_ID_STRING;
+import static org.sagebionetworks.bridge.BridgeUtils.AND_JOINER;
+import static org.sagebionetworks.bridge.BridgeUtils.isEmpty;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.annotation.Resource;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.dao.AssessmentDao;
+import org.sagebionetworks.bridge.models.PagedResourceList;
+import org.sagebionetworks.bridge.models.assessments.Assessment;
+import org.sagebionetworks.bridge.models.assessments.HibernateAssessment;
+
+@Component
+class HibernateAssessmentDao implements AssessmentDao {
+    static final String APP_ID = "appId";
+    static final String IDENTIFIER = "identifier";
+    static final String REVISION = "revision";
+    static final String GUID = "guid";
+        
+    static final String SELECT_COUNT = "SELECT COUNT(*)";
+    static final String SELECT_ALL = "SELECT *";
+    static final String GET_BY_IDENTIFIER = "FROM HibernateAssessment WHERE appId=:appId "+
+            "AND identifier=:identifier AND revision=:revision";
+    static final String GET_BY_GUID = "FROM HibernateAssessment WHERE appId=:appId AND guid=:guid";
+
+    static final String GET_REVISIONS = "FROM HibernateAssessment WHERE appId = :appId AND "
+            +"identifier = :identifier";
+    static final String GET_REVISIONS2 = "ORDER BY revision DESC";
+    static final String EXCLUDE_DELETED = "AND deleted = 0";
+    
+    private HibernateHelper hibernateHelper;
+    
+    @Resource(name = "basicHibernateHelper")
+    final void setHibernateHelper(HibernateHelper hibernateHelper) {
+        this.hibernateHelper = hibernateHelper;
+    }
+
+    @Override
+    public PagedResourceList<Assessment> getAssessments(String appId, int offsetBy, int pageSize,
+            Set<String> tags, boolean includeDeleted) {
+        
+        boolean includeTags = !isEmpty(tags);
+        
+        // Not sure pulling this out into constants is any easier to understand...
+        QueryBuilder builder = new QueryBuilder();
+        builder.append("FROM (");
+        builder.append("SELECT DISTINCT identifier as id, MAX(revision) AS rev FROM Assessments");
+        builder.append("GROUP BY identifier) AS latest_assessments");
+        builder.append("INNER JOIN Assessments AS a ON a.identifier = latest_assessments.id AND");
+        builder.append("a.revision = latest_assessments.rev");
+        
+        List<String> clauses = new ArrayList<>();
+        clauses.add("WHERE appId = :appId");
+        if (includeTags) {
+            clauses.add("guid IN (SELECT DISTINCT assessmentGuid FROM AssessmentTags WHERE tagValue IN :tags)");
+            //builder.getParameters().put("tags", tags);
+        }
```

Comment:
Corresponding unit test change?

Reply:
I restored this to its original form rather than writing a test... aesthetics aside there was no reason to change it.